### PR TITLE
feat: add getSwitches

### PR DESCRIPTION
**Pull request by sndrs**
##  What does this change?

adds the following:

- `getSwitches`

gets value from `window.guardian.config` if present, or from a new endpoint (`~~coming from~~ provided by @shtukas ❤️ ) if not.

also adds a couple of helpers:

- `isObject` and `isBoolean`

## Why?

- it will be super useful to be able to use these from other `@guardian/` packages
- https://github.com/orgs/guardian/teams/client-side-infra/discussions/10

### CODE REVIEW

File: src/getTests.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { isObject } from './isObject';
+import type { Tests } from './types/window';
+
+const URL = '';
```
Comment (mxdvl) [inline src/getTests.ts:4]: TBC

**Review by mxdvl — COMMENTED** [review]
Looking good!

**Comment by gtrufitt** [comment]
This looks great ⭐

File: src/getSwitches.ts [hunk 2]
```diff
@@ -1,16 +1,16 @@
 import { isBoolean } from './isBoolean';
 import { isObject } from './isObject';
-import type { Switches } from './types/window';
+import type { Switches } from './types/switches';
 
-const URL = '';
+const URL = 'https://www.theguardian.com/switches.json';
```
Comment (mxdvl) [inline src/getSwitches.ts:5]: That’s the one!

**Review by mxdvl — APPROVED** [review]
LGTM

**Review by mxdvl — DISMISSED** [review]
Excellent

**Comment by sndrs** [comment]
linking this for the next time this is picked up https://github.com/guardian/frontend/pull/23270/files

**Comment by sndrs** [comment]
@mxdvl this should be ready to merge now, does it still look ok to you?

**Comment by sndrs** [comment]
doh, no missing docs

**Comment by sndrs** [comment]
fixed

**Review by mxdvl — APPROVED** [review]
Yep, looking good! This will come in handy soon, I hope.

**Pull request merged**